Game.java

```diff
@@ -0,0 +1,109 @@
+import javax.swing.*;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class Game extends JFrame implements Runnable {
+
+    Board gameBoard; // the board the game is played on
+    AutoSolver solver; // the game automatic solver
+
+    int difficulty;
+    int solverSpeed; // how fast the solver is displayed
+    boolean buttonPressed;
+
+    public Game(int difficulty) {
+
+        JButton Check = new JButton("Check");
+        JButton AutoSolve = new JButton("Auto Solve");
+        JPanel options = new JPanel();
+        JSlider solverSpeed = new JSlider(DEF.MIN_VELOCITY, DEF.MAX_VELOCITY);
+
+        Check.setFocusable(false);
+        Check.addActionListener(new CheckListener());
+
+        AutoSolve.setFocusable(false);
+        AutoSolve.addActionListener(new AutoListener());
+
+        solverSpeed.setFocusable(false);
+        solverSpeed.addChangeListener(new SpeedListener());
+
+        GridLayout g = new GridLayout(1, 3);
+        options.setLayout(g);
+        options.add(Check);
+        options.add(solverSpeed);
+        options.add(AutoSolve);
+
+        buttonPressed = false;
+        int NumofHoles = 0;
+        switch (difficulty) {
+            case (1):
+                NumofHoles = 55;
+            case (2):
+                NumofHoles = 57;
+            case (3):
+                NumofHoles = 60;
+            case (4):
+                NumofHoles = 64;
+        }
+        
+        gameBoard = new Board(this, NumofHoles);
+        solver = new AutoSolver(gameBoard, this);
+        run();
+
+        this.add(gameBoard, BorderLayout.CENTER);
+        this.add(options, BorderLayout.SOUTH);
+
+        addKeyListener(gameBoard);
+        setFocusable(true);
+        setVisible(true);
+        setSize(DEF.GAME_HEIGHT, DEF.GAME_WIDTH);
+        setLocation(410, 105);
+
+    }
+
+    // listener to check all wrong/right fillings
+    private class CheckListener implements ActionListener {
+
+        public void actionPerformed(ActionEvent e) {
+
+            if (gameBoard.boardSolved(gameBoard.board)) {
+
+                setBackground(Color.green);
+            } else
+                gameBoard.check = !gameBoard.check;
+            repaint();
+        }
+
+    }
+
+    // listener to run the Auto solver
+    private class AutoListener implements ActionListener {
+
+        public synchronized void actionPerformed(ActionEvent e) {
+
+            solver.pause = !solver.pause;
+        }
+
+    }
+
+    // listener to determine solver speed
+    private class SpeedListener implements ChangeListener {
+
+        // determines how fast the solver runs
+        public void stateChanged(ChangeEvent e) {
+
+            JSlider x = (JSlider) e.getSource();
+            DEF.SOLVER_VELOCITY = x.getValue();

```

Comment:
Why change const value? You shouldn't do that

Reply:
not a const but a static var, used to determine all possible solvers speeds

Reply:
So move it from def to another location, def should contain only cons values(definitions which you indicate at the start of your program and will not change)